pageserver/src/config.rs

```diff
@@ -221,9 +225,11 @@ pub struct PageServerConf {
 
     pub tracing: Option<pageserver_api::config::Tracing>,
 
-    /// Enable TLS in page service API.
+    /// Enable TLS in page service API. This applies both to the libpq API and gRPC API.
     /// Does not force TLS: the client negotiates TLS usage during the handshake.
     /// Uses key and certificate from ssl_key_file/ssl_cert_file.
+    ///
+    /// TODO: consider making this unconditional for gRPC.
     pub enable_tls_page_service_api: bool,
```

Comment:
I'd be inclined to unconditionally use TLS for the gRPC API, to avoid a painful migration later. In that case, `enable_tls_page_service_api` would only have an effect for the libpq-based service.

Reply:
Well, I guess we still want to have `enable_tls_page_service_api` for tests/neon local, for example.
And if you decide to use TLS unconditionally, it means your API will not reach the production till TLS is enabled everywhere (we still are waiting for SRE team to build an infra for cert provisioning, and this task is not on their priority list right now)

Reply:
Another alternative would be to generate local-signed certificates using `rcgen` if there are none on the host. It will allow you to make tls unconditional and decouple from tls cert provisioning.
But you will need an option on client's side to disable cert validation in that case

Reply:
Thanks, we'll figure it out. Worst-case, we'll use a separate port for the TLS service.